### PR TITLE
[libc++][test] `ADDITIONAL_COMPILE_FLAGS` should be a space-separated list

### PR DESCRIPTION
**Pull request by StephanTLavavej**
Found while running libc++'s test suite with MSVC's STL.

`ADDITIONAL_COMPILE_FLAGS` is a `ParserKind.LIST`:

https://github.com/llvm/llvm-project/blob/3c23ed156f0151923b168bdff0c34ec73fb37f38/libcxx/utils/libcxx/test/format.py#L104-L108

With a comma-separated example:

https://github.com/llvm/llvm-project/blob/3c23ed156f0151923b168bdff0c34ec73fb37f38/libcxx/utils/libcxx/test/format.py#L223-L228

And comma-separated test coverage:

https://github.com/llvm/llvm-project/blob/dd3184c30ff531b8aecea280e65233337dd02815/libcxx/test/libcxx/selftest/additional_compile_flags/substitutes-in-run.sh.cpp#L12-L15

Because the machinery splits on commas:

https://github.com/llvm/llvm-project/blob/dd09221a29506031415cad8a1308998358633d48/llvm/utils/lit/lit/TestRunner.py#L1882-L1883
https://github.com/llvm/llvm-project/blob/dd09221a29506031415cad8a1308998358633d48/llvm/utils/lit/lit/TestRunner.py#L1951-L1956

However, most (although not all) usage of `ADDITIONAL_COMPILE_FLAGS` is treating it as space-separated. That apparently works in the normal Clang environment, but in my exotic configuration it causes `"-DMEOW -DWOOF"` to be passed as a single argument to MSVC, which then emits "warning C5102: ignoring invalid command-line macro definition `'_LIBCPP_DISABLE_DEPRECATION_WARNINGS -D_LIBCPP_ENABLE_CXX26_REMOVED_CODECVT'`", causing test failures due to warnings-as-errors.

This PR changes `ADDITIONAL_COMPILE_FLAGS` to actually be parsed as a space-separated list, and changes the few uses/examples that had commas.

Note: This gets my tests passing, and I tried to find everything I needed to change in the Python machinery, but I could have missed something. In particular, I tried to add good test coverage to `llvm/utils/lit/tests/unit/TestRunner.py` (intentionally verifying that commas are treated as normal characters) and `llvm/utils/lit/tests/Inputs/testrunner-custom-parsers/test.txt` but I'm uncertain whether that's actually being picked up because I found no other mentions of the previously existing `test_lists`.

### CODE REVIEW

**Review by mordante — APPROVED** [review]
TIL! Thanks for fixing these! I found 4 more in test/std, they all seem to be `-Xclang -verify-ignore-unexpected=error`. There are several more in libcxx. Feel free to fix these too, or leave a message, then I'll create a followup PR.

**Comment by StephanTLavavej** [comment]
@philnik777
> I think we should rather go the other way around and make the flag space-separated. Having it comma separated seems like a poor UI design to me, since every compiler I'm aware of assumes a space-separated list of arguments.

Space-separated seems like a good idea to me too - I didn't try that because it would be a large change to the test harness, and would involve changing a small amount of existing usage that's already using commas.

@mordante
> TIL! Thanks for fixing these! I found 4 more in test/std, they all seem to be -Xclang -verify-ignore-unexpected=error. There are several more in libcxx. Feel free to fix these too, or leave a message, then I'll create a followup PR.

You're welcome! If you could create a followup PR that would be great - I'm not equipped to properly test changes to the `-Xclang` options or anything in `libcxx/test/libcxx`.

**Comment by mordante** [comment]
> > TIL! Thanks for fixing these! I found 4 more in test/std, they all seem to be -Xclang -verify-ignore-unexpected=error. There are several more in libcxx. Feel free to fix these too, or leave a message, then I'll create a followup PR.
> 
> You're welcome! If you could create a followup PR that would be great

No problem, I'll do it after this patch has landed.

> - I'm not equipped to properly test changes to the `-Xclang` options or anything in `libcxx/test/libcxx`.

That's why we have a CI ;-)

**Comment by ldionne** [comment]
As a matter of process, I'd rather we fix all of the occurrences in this PR. We have CI in place that makes this easy, and maintainers can even push to this PR directly.

I also concur with Nikolas here, I think we should make it space-delimited instead.

File: llvm/utils/lit/lit/TestRunner.py
Comment (philnik777) [inline llvm/utils/lit/lit/TestRunner.py:1355]: This looks suspicious. Did zou change the whitespace somehow?

**Review by philnik777 — COMMENTED** [review]
LGTM, but Louis is a lot more knowledgable in his area, so I'd wait for his review.

File: llvm/utils/lit/lit/TestRunner.py
Comment (StephanTLavavej) [reply llvm/utils/lit/lit/TestRunner.py:1355]: Oh, my editor is configured to strip trailing whitespace on save. Would you like me to undo this line?

Comment (philnik777) [reply llvm/utils/lit/lit/TestRunner.py:1355]: Ah, that didn't show up on my phone. It just showed that you changes the line. That's fine then I think.

File: llvm/utils/lit/tests/Inputs/testrunner-custom-parsers/test.txt
Comment (ldionne) [inline llvm/utils/lit/tests/Inputs/testrunner-custom-parsers/test.txt:7]: We could add a test where we have multiple spaces between items.

**Review by ldionne — APPROVED** [review]
This is nice, thanks a lot for implementing this approach!

LGTM with a tiny test coverage suggestion.

File: llvm/utils/lit/tests/Inputs/testrunner-custom-parsers/test.txt
Comment (StephanTLavavej) [reply llvm/utils/lit/tests/Inputs/testrunner-custom-parsers/test.txt:7]: Great idea, pushed. This should be handled by my previous commit to avoid empty strings when splitting. (Without that logic, the initial space after the colon was emitting an empty string.)

**Pull request merged**
